code/GlobalNavSiteTreeExtension.php

```diff
@@ -8,8 +8,26 @@ class GlobalNavSiteTreeExtension extends DataExtension {
 	);
 
 
+	/**
+	 * Gets the hostname
+	 *
+	 * @return string Protocol and hostname only
+	 */
 	public static function get_toolbar_hostname() {
-		return Config::inst()->get('GlobalNav','use_localhost') ? Director::absoluteBaseURL() : Config::inst()->get('GlobalNav','hostname');
+		return Config::inst()->get('GlobalNav','use_localhost')
+			? Director::protocolAndHost()
```

Comment:
that's exactly the situation this code fixes; The link you gave would end up being 
`http://localhost/mysite/toolbar/css/toolbar.css`

Notice that that line doesn't call get_toolbar_hostname()... it calls get_toolbar_baseurl().